test_args.sh

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# build the binary
+go build cmd/livepeer/livepeer.go
+
+# set a clean slate "home dir" for testing
+TMPDIR=/tmp/livepeer-test-"$RANDOM"
+DEFAULT_DATADIR="$TMPDIR"/.lpData
+CUSTOM_DATADIR="$TMPDIR"/customDatadir
+export HOME=$TMPDIR
+rm -rf "$DEFAULT_DATADIR"
+mkdir -p $TMPDIR  # goclient should make the lpData datadir
+
+# Set up ethereum key
+cat > $TMPDIR/key <<ETH_KEY
+{"address":"089d8ab6752bac616a1f17246294eb068ee23d3e","crypto":{"cipher":"aes-128-ctr","ciphertext":"e868446e99842291b4991ae2c8e6c6834296c81937c4182e45af2edf0af61968","cipherparams":{"iv":"62205b25b7c4b2c35717128d9702f3c8"},"kdf":"scrypt","kdfparams":{"dklen":32,"n":262144,"p":1,"r":8,"salt":"dfa46628fec666ebd46d21d59cd3cbad0039d13f3c09dd5304128e32edfdec57"},"mac":"a5d4cde1863803a2b17718eb6b0770c0e7fcdfcf3659c9bd24f033d4529a5af3"},"id":"b84a7400-5fbd-4397-894e-c59403663b88","version":3}
+ETH_KEY
+
+ETH_ARGS="-ethKeystorePath $TMPDIR/key"
+SLEEP=0.25
+
+
+run_lp () {
+    ./livepeer "$@" &
+    pid=$!
+    sleep $SLEEP
+}
+
+# sanity check that default datadir does not exist
+[ ! -d "$DEFAULT_DATADIR" ]
+
+# check that we exit early if node type is not set
+res=0
+./livepeer || res=$?
+[ $res -ne 0 ]
+
+
+run_lp -broadcaster
+[ -d "$DEFAULT_DATADIR"/offchain ]
+kill $pid
+
+run_lp -broadcaster -network rinkeby $ETH_ARGS
+[ -d "$DEFAULT_DATADIR"/rinkeby ]
+kill $pid
+
+run_lp -broadcaster -network mainnet $ETH_ARGS
+[ -d "$DEFAULT_DATADIR"/mainnet ]
+kill $pid
+
+run_lp -broadcaster -network anyNetwork $ETH_ARGS -ethUrl "wss://rinkeby.infura.io/ws" -v 99
+[ -d "$DEFAULT_DATADIR"/anyNetwork ]
+kill $pid
+
+# sanity check that custom datadir does not exist
+[ ! -d "$CUSTOM_DATADIR" ]
+
+# check custom datadir without a network (offchain)
+run_lp -broadcaster -datadir "$CUSTOM_DATADIR"
+[ -d "$CUSTOM_DATADIR" ]
+[ ! -d  "$CUSTOM_DATADIR"/offchain ] # sanity check that network isn't included
+kill $pid
+
+# check custom datadir with a network
+run_lp -broadcaster -datadir "$CUSTOM_DATADIR" -network rinkeby $ETH_ARGS
+[ ! -d  "$CUSTOM_DATADIR"/rinkeby ] # sanity check that network isn't included
+kill $pid
+
+# check invalid / missing service address.
```

Comment:
More specifically, we'd want to test for a mismatch between the specified service address (either declared on-chain or user provided) and the inferred service address right?

Reply:
It's meant to ensure an unset serviceURI makes the availability check fail, since the inferred address will (most likely!) be non-public. Updated the comment to reflect that.

One day we should spend time getting this type of test to work reliably. Then there are various items we should test related to this specific item:
* Missing `serviceAddr` (the current test, but commented out)
* Malformed `serviceAddr` (added this test)
* Well-formed `serviceAddr` pointing somewhere else
* On-chain but empty `serviceURI`
* On-chain but malformed `serviceURI`
* On-chain, well-formed `serviceURI` pointing somewhere else